packages/blocks/src/page-block/edgeless/edgeless-page-block.ts

```diff
@@ -1062,10 +1065,35 @@ export class EdgelessPageBlockComponent
     this._resizeObserver = resizeObserver;
   }
 
+  private _initNoteResize() {
+    let raqId = 0;
+
+    const resetNoteResizeObserver = () => {
+      if (raqId) return;
+
+      raqId = requestAnimationFrame(() => {
+        this._noteResizeObserver.resetListener(this.page);
+        raqId = 0;
+      });
+    };
+
+    const listenChildrenUpdate = (root: BaseBlockModel<object> | null) => {
+      if (!root) return;
+
+      this._disposables.add(root.childrenUpdated.on(resetNoteResizeObserver));
+    };
+
+    listenChildrenUpdate(this.page.root);
+    this._disposables.add(
+      this.page.slots.rootAdded.on(root => listenChildrenUpdate(root))
+    );
+  }
+
   override firstUpdated() {
     this._initSlotEffects();
     this._initDragHandle();
     this._initResizeEffect();
+    this._initNoteResize();
```

Comment:
This is duplicate with the `_initResizeEffect` above. Can you try to get these 2 methods merged?

Reply:
Technically those two methods are not same. The name maybe a little confusion, I'll rename it.